nwbinspector/nwbinspector.py

```diff
@@ -184,52 +231,63 @@ def inspect_nwb(
         )
+
     file_name = Path(nwbfile_path).name
-    messages = list()
-    try:
-        with pynwb.NWBHDF5IO(path=str(nwbfile_path), mode="r", load_namespaces=True, driver=driver) as io:
-            validation_errors = pynwb.validate(io=io)
-            if any(validation_errors):
-                for validation_error in validation_errors:
-                    message = InspectorMessage(message=validation_error.reason)
-                    message.importance = Importance.PYNWB_VALIDATION
-                    message.check_function_name = validation_error.name
-                    message.location = validation_error.location
-                    message.file = file_name
-                    messages.append(message)
+    with pynwb.NWBHDF5IO(path=str(nwbfile_path), mode="r", load_namespaces=True, driver=driver) as io:
+        validation_errors = pynwb.validate(io=io)
+        if any(validation_errors):
+            for validation_error in validation_errors:
+                yield InspectorMessage(
+                    message=validation_error.reason,
+                    importance=Importance.PYNWB_VALIDATION,
+                    check_function_name=validation_error.name,
+                    location=validation_error.location,
+                    file=file_name,
+                )
+        try:
             nwbfile = io.read()
-            for check_function in checks:
-                if (
-                    (ignore is not None and check_function.__name__ in ignore)
-                    or (select is not None and check_function.__name__ not in select)
-                    or (check_function.importance.value < importance_threshold.value)
-                ):
-                    continue
-                for nwbfile_object in nwbfile.objects.values():
-                    if issubclass(type(nwbfile_object), check_function.neurodata_type):
-                        check_result = check_function(nwbfile_object)
-                        if check_result is not None:
-                            if isinstance(check_result, Iterable):
-                                for message in check_result:
-                                    message.file = file_name
-                                    messages.append(message)
-                            else:
-                                check_result.file = file_name
-                                messages.append(check_result)
-    except Exception as ex:
-        message = InspectorMessage(message=traceback.format_exc())
-        message.importance = Importance.ERROR
-        message.check_function_name = f"{type(ex)}: {str(ex)}"
-        message.file = file_name
-        messages.append(message)
-    return messages
+        except Exception as ex:
+            yield InspectorMessage(
+                message=traceback.format_exc(),
+                importance=Importance.ERROR,
+                check_function_name=f"{type(ex)}: {str(ex)}",
```

Comment:
This type of change was needed when using the new organization method with one of the levels being `check_function_name`.